### PR TITLE
Add development quickstart docs

### PR DESCRIPTION
**Pull request by kragniz**
Mostly for my own use, but might be handy for other people new to the
project. Feel free to suggest other workflows.

/kind documentation

```release-note
NONE
```

### CODE REVIEW

File: docs/developing.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    helm install contrib/charts/navigator \
+        --set apiserver.image.pullPolicy=IfNotPresent \
+        --set apiserver.image.tag=dev \
+        --set controller.image.pullPolicy=IfNotPresent \
+        --set controller.image.tag=dev \
+        --name navigator --namespace navigator --wait
+
+Test navigator is deployed by creating a demo elasticsearch cluster:
+
+    kubectl create -f docs/quick-start/es-cluster-demo.yaml
```
Comment (munnerz) [inline docs/developing.md]: Worth adding here that you should set the pilot image tag to `dev`, and set pull policy to Never as well.

File: docs/developing.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+Install minikube and start a cluster with RBAC enabled:
+
+    minikube start --extra-config=apiserver.Authorization.Mode=RBAC
+
+Work around `kube-dns` and other services having problems when RBAC is enabled in minikube:
+
+    kubectl create clusterrolebinding cluster-admin:kube-system \
+        --clusterrole=cluster-admin \
+        --serviceaccount=kube-system:default
```
[the remark this answers is not in the document]
Comment (kragniz) [reply docs/developing.md]: 👍

File: docs/developing.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+Fetch the docker configuration:
+
+    eval (minikube docker-env)
```
[the remark this answers is not in the document]
Comment (kragniz) [reply docs/developing.md]: this is correct for me (in fish), but I forgot to change it for bash/zsh

File: docs/developing.md (hunk 1 above)
Comment (kragniz) [reply docs/developing.md]: :+1:

**Pull request merged**